File: src/StringAndArray/Exercise452.java
package StringAndArray;

import java.util.Arrays;

public class Exercise452 {
    class Solution {
        public int findMinArrowShots(int[][] points) {
            Arrays.sort(points, (a, b) -> Integer.compare(a[1], b[1]));
            int lo = points[0][0];
            int hi = points[0][1];
            int count = 1;
            for (int i = 1; i < points.length; i++) {
                if (points[i][0] <= hi)
                    lo = (lo < points[i][0] ? points[i][0] : lo);

                else {
                    count++;
                    lo = points[i][0];
                    hi = points[i][1];
                }

            }
            return count;
        }
    }
    public static void main(String[] args) {
        Exercise452 exercise = new Exercise452();
        Solution solution = exercise.new Solution();
        int[][] nums = new int[][] { { 10, 16 }, { 2, 8 }, { 1, 6 }, { 7, 12 } };
        System.out.println(solution.findMinArrowShots(nums));
    }
}

File: src/Intervals/Exercise452.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution
{
public:
    int findMinArrowShots(vector<vector<int>> &points)
    {
        sort(points.begin(), points.end(), [](const vector<int> &a, const vector<int> &b)
             { return a[0] < b[0]; });

        int count = 1;
        int left = points[0][0], right = points[0][1];
        for (int i = 1; i < points.size(); i++)
        {
            if (points[i][0] > right)
            {
                count++;
                left = points[i][0];
                right = points[i][1];
            }
            else
            {
                left = points[i][0];
                right = min(points[i][1], right);
            }
        }
        return count;
    }
};

int main()
{
    vector<vector<int>> points = {{10, 16}, {2, 8}, {1, 6}, {7, 12}};
    Solution sl;
    cout << sl.findMinArrowShots(points);
}